### DISCUSSION TITLE
Android client with openssl 3.0.7???

### DISCUSSION THREAD

**Discussion by jlensaz** [strongswan/strongswan#1492] [Q&A]
Has anyone successfully built the strongswan android client openssl 3.0.7 in Android Studio?

I'm using strongswan v5.9.9 and android NDK 25.1.8937393.  I can successfully build with SS 5.9.9, NDK 22.x, and boringSSL but we are being pushed to the latest openSSL for security patch updates.

The error I'm getting is:  Android NDK: Module strongswan depends on undefined modules: crypto_static

**Comment by tobiasbrunner** [comment]
You obviously have to adapt the build for that (e.g. similar to [our F-Droid build](https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/org.strongswan.android.yml)).

**Reply by rohit8878** [reply]
Hi @tobiasbrunner Can you please provide any document, how can i do the F-Droid configuration integration

**Reply by tobiasbrunner** [reply]
The switch from BoringSSL to OpenSSL has been done in the meantime, see the [current build instructions](https://docs.strongswan.org/docs/5.9/os/androidVpnClientBuild.html).
